app/src/main/java/com/github/sdp_begreen/begreen/activities/MainActivity.kt

```diff
@@ -183,12 +183,10 @@ class MainActivity : AppCompatActivity() {
                 item.setIcon(R.drawable.ic_baseline_feed)
 
                 // Feed posts
-                // TODO : For now, it displays the user own posts.
-                val user = connectedUserViewModel.currentUser.value
-
-                val photos = user?.trashPhotosMetadatasList ?: listOf()
-                replaceFragInMainContainer(UserPhotoFragment.newInstance(1, photos, true))
+                val feedPosts = runBlocking { fetchFeedPosts() }
```

Comment:
We should probably use `lifecycleScope.launch` instead of `runBlocking` to be consistent with the rest of the code

Reply:
Actually, I took inspiration of the rest of the class to be consistent with it. It seems that other methods of the `MainActivity` use `runBlocking` in the same way, before calling `replaceFragInMainContainer`.

<img width="877" alt="image" src="https://github.com/SDP-BeGreen/BeGreen/assets/91261122/f5a9fd10-a2bd-47b8-ba5b-fb88929f13c3">

It seems to not work with `lifecycleScope.launch` since it returns a job and not the desired list :

<img width="784" alt="image" src="https://github.com/SDP-BeGreen/BeGreen/assets/91261122/ae08fdc1-36ee-4ae2-8edb-370460f53c19">

The only way to use `lifecycleScope.launch` is to embed the `replaceFragInMainContainer` inside it and it works fine. However, it will not be the same as the rest of the class. What do you think about it ?

Reply:
That's true, most of this code was written long time ago and we were not using this already. Your error comes from the fact that you should also put the line below in the `lifecycleScope.launch`, but its up to you to do this refactor to the rest of the class or not

Reply:
For the reason you are telling, I preferred to use `lifecycleScope.launch` as you said, and I applied the changes in the rest of the class as well. Thank you for this relevant suggestion.